eip-0050.md

```diff
@@ -0,0 +1,112 @@
+Sigma 6.0
+=========
+
+This EIP contains proposed changes for Ergo contractual layer, called Sigma. These changes are based on found issues 
+in the current version of Sigma, as well as feedback got from users, i.e. Ergo ecosystem developers.
+
+
+Activation 
+----------
+
+Rule 1011 should be replaced with another one, likely with the same rule but under different id.
+
+Features Set 
+------------
+
+* *.validatePoW* method of *Header* type to validate Autolykos2 proof of work done on header 
+
+issue: https://github.com/ScorexFoundation/sigmastate-interpreter/issues/958
+
+* implement conversion from Long-encoded nBits representation to BigInt (long.fromNBits method) and to nBits from BitInt (bigInt.toNBits method) 
+
+issue: https://github.com/ScorexFoundation/sigmastate-interpreter/issues/675
+
+PR: https://github.com/ScorexFoundation/sigmastate-interpreter/pull/962 (nbits encoding)
+
+
+* implement Boolean to Byte conversion (boolean.toByte method)
+
+issue: https://github.com/ScorexFoundation/sigmastate-interpreter/issues/931
+PR: https://github.com/ScorexFoundation/sigmastate-interpreter/pull/932
+
+* implement BigIntModQ type (unsigned bigints mod Q, where Q is the order of secp256k1 group)
+
+issue: https://github.com/ScorexFoundation/sigmastate-interpreter/issues/554
+
+
+* finish executeFromVar implementation
+
+issue: https://github.com/ScorexFoundation/sigmastate-interpreter/issues/443 , https://github.com/ScorexFoundation/sigmastate-interpreter/issues/612
+
+* support MethodCall encoding of Numeric methods
+
+issue: https://github.com/ScorexFoundation/sigmastate-interpreter/issues/667
```

Comment:
Maybe this is not necessary. Looking back is was not good idea to allow both `method call` and `opcode` encoding for methods. It would be better to forbid method calls for those methods which have dedicated opcodes.
The translation MC -> opcode is actually happens in the compiler, but is not strictly enforced. This creates more surface for test coverage and potential bugs.